demos/dotnet/dotnet/dotnet.cs

```diff
@@ -157,8 +180,24 @@ public static ProjectProperties InitializeProperties(string[] args, Log log, boo
         properties.OutputType = buildDll ? ".dll" : ".exe";
         FindCompiler(properties);
 
+        var projectFiles = Directory.GetFiles(properties.ProjectDirectory, "*.dotnetproj");
+        bool projectFileExists = projectFiles.Length > 0;
         // Sources
-        properties.Sources.AddRange(Directory.GetFiles(properties.ProjectDirectory, "*.cs"));
+        if (projectFileExists)
+        {
+            foreach(var projectFile in projectFiles)
+            {
+                properties.Sources.AddRange(ParseProjectFile(properties.ProjectDirectory, projectFile));
+            }
+        }
+        properties.Sources.AddRange(Directory.GetFiles(properties.ProjectDirectory, "*.cs", (args.Length > 1 && args[1] == "*.cs") ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly));
+        if (args.Length > 1 && args[1] != "*.cs")
```

Comment:
Let's say the command argument is generally of the form "/*:*". Once we split the argument and the specification (split over ':'), we have two elements only - the command option and the specification. Once we grab the specification (i.e. whether it is '*.cs' or a subdirectory name), we can get the appropriate source files.